sys-devel/llvm/llvm-19.0.0.9999.ebuild

```diff
@@ -538,6 +560,16 @@ multilib_src_install_all() {
 }
 
 pkg_postinst() {
+	# Always load LLVMPolly.so so -Xclang -load -Xclang LLVMPolly.so is not needed to use Polly
+	# It is harmless if polly optimizations are not used
+	if use polly; then
+		local LLVM_libdir="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/$(get_libdir)"
+		local libLLVM=$(find "${LLVM_libdir}" -maxdepth 1 -type f -name 'libLLVM.so*' | tail -n1)
+
+		einfo "Patching $(basename ${libLLVM}) to automatically load LLVMPolly.so ..."
+		patchelf --add-needed "LLVMPolly.so" "${libLLVM}"
+	fi
```

Comment:
That's definitely *not* something to be done in postinst.

Also, I'm wondering if there's a point in installing it as a shared library if you always link it anyway, i.e. if we shouldn't just do `LINK_POLLY_INTO_TOOLS`, though I don't know if that's going to work with clang.

That said, I'd really prefer having a separate ebuild for Polly, installing it as a dynamic plugin and loading it via configs from `sys-devel/clang-common`.

Reply:
I'd also prefer we do that separately and not tie it in with the other issues here.

Reply:
> That's definitely _not_ something to be done in postinst.
> 
> Also, I'm wondering if there's a point in installing it as a shared library if you always link it anyway, i.e. if we shouldn't just do `LINK_POLLY_INTO_TOOLS`, though I don't know if that's going to work with clang.
> 
> That said, I'd really prefer having a separate ebuild for Polly, installing it as a dynamic plugin and loading it via configs from `sys-devel/clang-common`.

The problem with `LINK_POLLY_INTO_TOOLS` is that it forces you to define a lot of static targets in distribution and also to install a lot of .a files or otherwise clang will fail after to compile (this is what was happening in my last attempts)

Your clang-common aproach also seems reasonable

Reply:
I've tried manually and the clang-common (with -Xclang -load -Xclang LLVMPolly..so) aproach works.

Should I force-push a new version with external polly and clang-common changes?

Reply:
Yes, please. Also one commit per package, please.